apps/mobile/src/libs/components/gradient-slider/gradient-slider.tsx

```diff
@@ -0,0 +1,134 @@
+import Slider from "@react-native-community/slider";
+import React from "react";
+
+import {
+	Icon,
+	LinearGradient,
+	SliderSection,
+	Text,
+	View,
+} from "~/libs/components/components";
+import { BaseColor } from "~/libs/enums/enums";
+import {
+	generateSliderGradientColors,
+	generateSliderGradientLocations,
+} from "~/libs/helpers/helpers";
+import { useCallback, useState } from "~/libs/hooks/hooks";
+import {
+	type colorToGradientColors,
+	directionToGradient,
+} from "~/libs/maps/maps";
+import { globalStyles } from "~/libs/styles/styles";
+import { type LayoutChangeEvent } from "~/libs/types/types";
+
+import { SliderConfig, SliderStyle } from "./libs/enums/enums";
+import { styles } from "./styles";
+
+type Properties = {
+	gradientColors: keyof typeof colorToGradientColors;
+	max: number;
+	min: number;
+	onValueChange?: (value: number) => void;
+};
+
+const GradientSlider: React.FC<Properties> = ({
+	gradientColors,
+	max,
+	min,
+	onValueChange,
+}) => {
+	const [value, setValue] = useState<number>(min);
+	const [sliderWidth, setSliderWidth] = useState<number>(
+		SliderConfig.SLIDER_INITIAL_WIDTH,
+	);
+	const initialColors = generateSliderGradientColors(min, gradientColors, max);
+	const [color, setColor] = useState<string[]>(initialColors);
+
+	const handleLayout = useCallback((event: LayoutChangeEvent): void => {
+		const { width } = event.nativeEvent.layout;
+		setSliderWidth(width);
+	}, []);
+
+	const handleValueChange = useCallback(
+		(sliderValue: number) => {
+			setValue(sliderValue);
+			setColor(generateSliderGradientColors(sliderValue, gradientColors, max));
+
+			if (onValueChange) {
+				onValueChange(sliderValue);
+			}
```

Comment:
```suggestion
				onValueChange?.(sliderValue);
```
Prefer modern JS syntax

Reply:
change it